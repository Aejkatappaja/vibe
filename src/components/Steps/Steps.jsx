import React, { useRef, forwardRef } from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import {
  STEPS_NUMBERS_TYPE,
  STEPS_GALLERY_TYPE,
  STEPS_CSS_BASE_CLASS,
  MAX_STEPS_FOR_GALLERY_TYPE
} from "./StepsConstants";
import useMergeRefs from "../../hooks/useMergeRefs";
import { StepsHeader } from "./StepsHeader";
import "./Steps.scss";
import { NOOP } from "../../utils/function-utils";
import { BEMClass } from "../../helpers/bem-helper";

const CSS_BASE_CLASS = `${STEPS_CSS_BASE_CLASS}-wrapper`;
const bemHelper = BEMClass(CSS_BASE_CLASS);

const Steps = forwardRef(({ className, id, steps, activeStepIndex, type, onChangeActiveStep, isOnPrimary }, ref) => {
  const componentRef = useRef(null);
  const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
  const overrideType = steps.length > MAX_STEPS_FOR_GALLERY_TYPE ? STEPS_NUMBERS_TYPE : type;
  return (
    <div
      ref={mergedRef}
      className={cx(CSS_BASE_CLASS, className, { [bemHelper({ state: "on-primary" })]: isOnPrimary })}
      id={id}
    >
      <StepsHeader
        onChangeActiveStep={onChangeActiveStep}
        type={overrideType}
        activeStepIndex={activeStepIndex}
        stepsCount={steps.length}
      />
      {steps[activeStepIndex]}
    </div>
  );
});

Steps.types = [STEPS_NUMBERS_TYPE, STEPS_GALLERY_TYPE];

Steps.propTypes = {
  activeStepIndex: PropTypes.number,
  steps: PropTypes.arrayOf(PropTypes.element),
  className: PropTypes.string,
  id: PropTypes.string,
  type: PropTypes.oneOf([STEPS_GALLERY_TYPE, STEPS_NUMBERS_TYPE]),
  isOnPrimary: PropTypes.bool
};

Steps.defaultProps = {
  activeStepIndex: 1,
  isOnPrimary: false,
  steps: [],
  className: "",
  id: "",
  onChangeActiveStep: NOOP,
  type: STEPS_GALLERY_TYPE
};

export default Steps;
