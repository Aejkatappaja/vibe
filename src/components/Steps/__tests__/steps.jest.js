import React from "react";
import { fireEvent, render, cleanup } from "@testing-library/react";
import { act } from "@testing-library/react-hooks";
import Steps from "../Steps";
import renderer from "react-test-renderer";
import { sinon } from "../../../test/test-helpers";
import { BACK_COMMAND_TEST_ID, NEXT_COMMAND_TEST_ID } from "../StepsConstants";

jest.useFakeTimers();
const steps = [<div data-testid="first-step">first</div>, <div data-testid="second-step">second</div>];
const renderComponent = ({ ...props }) => {
  return render(<Steps steps={steps} {...props}></Steps>);
};
describe("Steps tests", () => {
  describe("Snapshot Tests", () => {
    it("renders correctly with empty props", () => {
      const tree = renderer.create(<Steps />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    describe("renders correctly with numbers view", () => {
      it("with regular props", () => {
        const tree = renderer.create(<Steps type={Steps.types.NUMBERS} steps={steps} />).toJSON();
        expect(tree).toMatchSnapshot();
      });
      it("when viewing first step", () => {
        const tree = renderer.create(<Steps type={Steps.types.NUMBERS} steps={steps} activeStepIndex={0} />).toJSON();
        expect(tree).toMatchSnapshot();
      });
      it("when viewing last step", () => {
        const tree = renderer
          .create(<Steps type={Steps.types.NUMBERS} steps={steps} activeStepIndex={steps.length - 1} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
    describe("renders correctly with gallery view", () => {
      it("with regular props", () => {
        const tree = renderer.create(<Steps type={Steps.types.GALLERY} steps={steps} />).toJSON();
        expect(tree).toMatchSnapshot();
      });
      it("when viewing first step", () => {
        const tree = renderer.create(<Steps type={Steps.types.GALLERY} steps={steps} activeStepIndex={0} />).toJSON();
        expect(tree).toMatchSnapshot();
      });
      it("when viewing last step", () => {
        const tree = renderer.create(
          <Steps type={Steps.types.GALLERY} steps={steps} activeStepIndex={steps.length - 1} />
        );
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe("Integration Tests", () => {
    afterEach(() => {
      cleanup();
    });

    it("call onChangeIndexCallback when click on go back button and it does not disable", () => {
      const onClickMock = jest.fn();
      const steps = renderComponent({
        onChangeActiveStep: onClickMock,
        activeStepIndex: steps.length - 1
      });
      const backwardButton = steps.getByTestId(BACK_COMMAND_TEST_ID);

      act(() => {
        fireEvent.click(backwardButton);
      });

      expect(onClickMock.mock.calls.length).toBe(1);
    });
    it("call onChangeIndexCallback when click on go back button and it does not disable", () => {
      const onClickMock = jest.fn();
      const steps = renderComponent({
        onChangeActiveStep: onClickMock,
        activeStepIndex: 0
      });
      const forwardButton = steps.getByTestId(NEXT_COMMAND_TEST_ID);

      act(() => {
        fireEvent.click(forwardButton);
      });

      expect(onClickMock.mock.calls.length).toBe(1);
    });
    it("does not call onChangeIndexCallback when click on go back button and it disable", () => {
      const onClickMock = jest.fn();
      const steps = renderComponent({
        onChangeActiveStep: onClickMock,
        activeStepIndex: 0
      });
      const backwardButton = steps.getByTestId(BACK_COMMAND_TEST_ID);

      act(() => {
        fireEvent.click(backwardButton);
      });

      expect(onClickMock.mock.calls.length).toBe(0);
    });
    it("does not call onChangeIndexCallback when click on go back button and it disable", () => {
      const onClickMock = jest.fn();
      const steps = renderComponent({
        onChangeActiveStep: onClickMock,
        activeStepIndex: steps.length - 1
      });
      const forwardButton = steps.getByTestId(NEXT_COMMAND_TEST_ID);

      act(() => {
        fireEvent.click(forwardButton);
      });

      expect(onClickMock.mock.calls.length).toBe(0);
    });
  });
});
