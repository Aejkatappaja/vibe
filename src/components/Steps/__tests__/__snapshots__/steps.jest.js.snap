// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps tests Snapshot Tests renders correctly with empty props 1`] = `
<div
  className="monday-style-steps"
  id=""
>
  <div
    className="monday-style-steps-header monday-style-steps-header--gallery"
  >
    <div
      className="monday-style-steps-command monday-style-steps-command--backward"
      data-testid="steps-backward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Back
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M5.46967 10.5303L6 10L5.46967 9.46967C5.17678 9.76256 5.17678 10.2374 5.46967 10.5303ZM7.06066 10L13.5303 3.53033C13.8232 3.23744 13.8232 2.76256 13.5303 2.46967C13.2374 2.17678 12.7626 2.17678 12.4697 2.46967L5.46967 9.46967L6 10L5.46967 10.5303L12.4697 17.5303C12.7626 17.8232 13.2374 17.8232 13.5303 17.5303C13.8232 17.2374 13.8232 16.7626 13.5303 16.4697L7.06066 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="monday-style-steps-header_gallery"
    />
    <div
      className="monday-style-steps-command monday-style-steps-command--forward"
      data-testid="steps-forward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Next
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M13.5303 9.46967L13 10L13.5303 10.5303C13.8232 10.2374 13.8232 9.76256 13.5303 9.46967ZM11.9393 10L5.46967 16.4697C5.17678 16.7626 5.17678 17.2374 5.46967 17.5303C5.76256 17.8232 6.23744 17.8232 6.53033 17.5303L13.5303 10.5303L13 10L13.5303 9.46967L6.53033 2.46967C6.23744 2.17678 5.76256 2.17678 5.46967 2.46967C5.17678 2.76256 5.17678 3.23744 5.46967 3.53033L11.9393 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`Steps tests Snapshot Tests renders correctly with gallery view when viewing first step 1`] = `
<div
  className="monday-style-steps"
  id=""
>
  <div
    className="monday-style-steps-header monday-style-steps-header--gallery"
  >
    <div
      className="monday-style-steps-command monday-style-steps-command--backward"
      data-testid="steps-backward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Back
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M5.46967 10.5303L6 10L5.46967 9.46967C5.17678 9.76256 5.17678 10.2374 5.46967 10.5303ZM7.06066 10L13.5303 3.53033C13.8232 3.23744 13.8232 2.76256 13.5303 2.46967C13.2374 2.17678 12.7626 2.17678 12.4697 2.46967L5.46967 9.46967L6 10L5.46967 10.5303L12.4697 17.5303C12.7626 17.8232 13.2374 17.8232 13.5303 17.5303C13.8232 17.2374 13.8232 16.7626 13.5303 16.4697L7.06066 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="monday-style-steps-header_gallery"
    >
      <div
        className="monday-style-steps-header_dot monday-style-steps-header_dot--is-active"
        onClick={[Function]}
        onDragStart={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      />
      <div
        className="monday-style-steps-header_dot"
        onClick={[Function]}
        onDragStart={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      />
    </div>
    <div
      className="monday-style-steps-command monday-style-steps-command--forward"
      data-testid="steps-forward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Next
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M13.5303 9.46967L13 10L13.5303 10.5303C13.8232 10.2374 13.8232 9.76256 13.5303 9.46967ZM11.9393 10L5.46967 16.4697C5.17678 16.7626 5.17678 17.2374 5.46967 17.5303C5.76256 17.8232 6.23744 17.8232 6.53033 17.5303L13.5303 10.5303L13 10L13.5303 9.46967L6.53033 2.46967C6.23744 2.17678 5.76256 2.17678 5.46967 2.46967C5.17678 2.76256 5.17678 3.23744 5.46967 3.53033L11.9393 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
  <div
    data-testid="first-step"
  >
    first
  </div>
</div>
`;

exports[`Steps tests Snapshot Tests renders correctly with gallery view when viewing last step 1`] = `
<div
  className="monday-style-steps"
  id=""
>
  <div
    className="monday-style-steps-header monday-style-steps-header--gallery"
  >
    <div
      className="monday-style-steps-command monday-style-steps-command--backward"
      data-testid="steps-backward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Back
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M5.46967 10.5303L6 10L5.46967 9.46967C5.17678 9.76256 5.17678 10.2374 5.46967 10.5303ZM7.06066 10L13.5303 3.53033C13.8232 3.23744 13.8232 2.76256 13.5303 2.46967C13.2374 2.17678 12.7626 2.17678 12.4697 2.46967L5.46967 9.46967L6 10L5.46967 10.5303L12.4697 17.5303C12.7626 17.8232 13.2374 17.8232 13.5303 17.5303C13.8232 17.2374 13.8232 16.7626 13.5303 16.4697L7.06066 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="monday-style-steps-header_gallery"
    >
      <div
        className="monday-style-steps-header_dot"
        onClick={[Function]}
        onDragStart={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      />
      <div
        className="monday-style-steps-header_dot monday-style-steps-header_dot--is-active"
        onClick={[Function]}
        onDragStart={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      />
    </div>
    <div
      className="monday-style-steps-command monday-style-steps-command--forward"
      data-testid="steps-forward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Next
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M13.5303 9.46967L13 10L13.5303 10.5303C13.8232 10.2374 13.8232 9.76256 13.5303 9.46967ZM11.9393 10L5.46967 16.4697C5.17678 16.7626 5.17678 17.2374 5.46967 17.5303C5.76256 17.8232 6.23744 17.8232 6.53033 17.5303L13.5303 10.5303L13 10L13.5303 9.46967L6.53033 2.46967C6.23744 2.17678 5.76256 2.17678 5.46967 2.46967C5.17678 2.76256 5.17678 3.23744 5.46967 3.53033L11.9393 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
  <div
    data-testid="second-step"
  >
    second
  </div>
</div>
`;

exports[`Steps tests Snapshot Tests renders correctly with gallery view with regular props 1`] = `
<div
  className="monday-style-steps"
  id=""
>
  <div
    className="monday-style-steps-header monday-style-steps-header--gallery"
  >
    <div
      className="monday-style-steps-command monday-style-steps-command--backward"
      data-testid="steps-backward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Back
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M5.46967 10.5303L6 10L5.46967 9.46967C5.17678 9.76256 5.17678 10.2374 5.46967 10.5303ZM7.06066 10L13.5303 3.53033C13.8232 3.23744 13.8232 2.76256 13.5303 2.46967C13.2374 2.17678 12.7626 2.17678 12.4697 2.46967L5.46967 9.46967L6 10L5.46967 10.5303L12.4697 17.5303C12.7626 17.8232 13.2374 17.8232 13.5303 17.5303C13.8232 17.2374 13.8232 16.7626 13.5303 16.4697L7.06066 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="monday-style-steps-header_gallery"
    >
      <div
        className="monday-style-steps-header_dot"
        onClick={[Function]}
        onDragStart={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      />
      <div
        className="monday-style-steps-header_dot monday-style-steps-header_dot--is-active"
        onClick={[Function]}
        onDragStart={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      />
    </div>
    <div
      className="monday-style-steps-command monday-style-steps-command--forward"
      data-testid="steps-forward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Next
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M13.5303 9.46967L13 10L13.5303 10.5303C13.8232 10.2374 13.8232 9.76256 13.5303 9.46967ZM11.9393 10L5.46967 16.4697C5.17678 16.7626 5.17678 17.2374 5.46967 17.5303C5.76256 17.8232 6.23744 17.8232 6.53033 17.5303L13.5303 10.5303L13 10L13.5303 9.46967L6.53033 2.46967C6.23744 2.17678 5.76256 2.17678 5.46967 2.46967C5.17678 2.76256 5.17678 3.23744 5.46967 3.53033L11.9393 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
  <div
    data-testid="second-step"
  >
    second
  </div>
</div>
`;

exports[`Steps tests Snapshot Tests renders correctly with numbers view when viewing first step 1`] = `
<div
  className="monday-style-steps"
  id=""
>
  <div
    className="monday-style-steps-header monday-style-steps-header--numbers"
  >
    <div
      className="monday-style-steps-command monday-style-steps-command--backward"
      data-testid="steps-backward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Back
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M5.46967 10.5303L6 10L5.46967 9.46967C5.17678 9.76256 5.17678 10.2374 5.46967 10.5303ZM7.06066 10L13.5303 3.53033C13.8232 3.23744 13.8232 2.76256 13.5303 2.46967C13.2374 2.17678 12.7626 2.17678 12.4697 2.46967L5.46967 9.46967L6 10L5.46967 10.5303L12.4697 17.5303C12.7626 17.8232 13.2374 17.8232 13.5303 17.5303C13.8232 17.2374 13.8232 16.7626 13.5303 16.4697L7.06066 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="monday-style-steps-header_numbers"
    >
      1 \\ 2
    </div>
    <div
      className="monday-style-steps-command monday-style-steps-command--forward"
      data-testid="steps-forward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Next
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M13.5303 9.46967L13 10L13.5303 10.5303C13.8232 10.2374 13.8232 9.76256 13.5303 9.46967ZM11.9393 10L5.46967 16.4697C5.17678 16.7626 5.17678 17.2374 5.46967 17.5303C5.76256 17.8232 6.23744 17.8232 6.53033 17.5303L13.5303 10.5303L13 10L13.5303 9.46967L6.53033 2.46967C6.23744 2.17678 5.76256 2.17678 5.46967 2.46967C5.17678 2.76256 5.17678 3.23744 5.46967 3.53033L11.9393 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
  <div
    data-testid="first-step"
  >
    first
  </div>
</div>
`;

exports[`Steps tests Snapshot Tests renders correctly with numbers view when viewing last step 1`] = `
<div
  className="monday-style-steps"
  id=""
>
  <div
    className="monday-style-steps-header monday-style-steps-header--numbers"
  >
    <div
      className="monday-style-steps-command monday-style-steps-command--backward"
      data-testid="steps-backward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Back
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M5.46967 10.5303L6 10L5.46967 9.46967C5.17678 9.76256 5.17678 10.2374 5.46967 10.5303ZM7.06066 10L13.5303 3.53033C13.8232 3.23744 13.8232 2.76256 13.5303 2.46967C13.2374 2.17678 12.7626 2.17678 12.4697 2.46967L5.46967 9.46967L6 10L5.46967 10.5303L12.4697 17.5303C12.7626 17.8232 13.2374 17.8232 13.5303 17.5303C13.8232 17.2374 13.8232 16.7626 13.5303 16.4697L7.06066 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="monday-style-steps-header_numbers"
    >
      2 \\ 2
    </div>
    <div
      className="monday-style-steps-command monday-style-steps-command--forward"
      data-testid="steps-forward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Next
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M13.5303 9.46967L13 10L13.5303 10.5303C13.8232 10.2374 13.8232 9.76256 13.5303 9.46967ZM11.9393 10L5.46967 16.4697C5.17678 16.7626 5.17678 17.2374 5.46967 17.5303C5.76256 17.8232 6.23744 17.8232 6.53033 17.5303L13.5303 10.5303L13 10L13.5303 9.46967L6.53033 2.46967C6.23744 2.17678 5.76256 2.17678 5.46967 2.46967C5.17678 2.76256 5.17678 3.23744 5.46967 3.53033L11.9393 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
  <div
    data-testid="second-step"
  >
    second
  </div>
</div>
`;

exports[`Steps tests Snapshot Tests renders correctly with numbers view with regular props 1`] = `
<div
  className="monday-style-steps"
  id=""
>
  <div
    className="monday-style-steps-header monday-style-steps-header--numbers"
  >
    <div
      className="monday-style-steps-command monday-style-steps-command--backward"
      data-testid="steps-backward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Back
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M5.46967 10.5303L6 10L5.46967 9.46967C5.17678 9.76256 5.17678 10.2374 5.46967 10.5303ZM7.06066 10L13.5303 3.53033C13.8232 3.23744 13.8232 2.76256 13.5303 2.46967C13.2374 2.17678 12.7626 2.17678 12.4697 2.46967L5.46967 9.46967L6 10L5.46967 10.5303L12.4697 17.5303C12.7626 17.8232 13.2374 17.8232 13.5303 17.5303C13.8232 17.2374 13.8232 16.7626 13.5303 16.4697L7.06066 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="monday-style-steps-header_numbers"
    >
      2 \\ 2
    </div>
    <div
      className="monday-style-steps-command monday-style-steps-command--forward"
      data-testid="steps-forward-command"
      onClick={[Function]}
      onDragStart={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="monday-style-steps-command_text"
      >
        Next
      </span>
      <svg
        aria-hidden={true}
        className="icon_component monday-style-steps-command_icon"
        fill="currentColor"
        height="16"
        onClick={[Function]}
        role="img"
        viewBox="0 0 20 20"
        width="16"
      >
        <path
          clipRule="evenodd"
          d="M13.5303 9.46967L13 10L13.5303 10.5303C13.8232 10.2374 13.8232 9.76256 13.5303 9.46967ZM11.9393 10L5.46967 16.4697C5.17678 16.7626 5.17678 17.2374 5.46967 17.5303C5.76256 17.8232 6.23744 17.8232 6.53033 17.5303L13.5303 10.5303L13 10L13.5303 9.46967L6.53033 2.46967C6.23744 2.17678 5.76256 2.17678 5.46967 2.46967C5.17678 2.76256 5.17678 3.23744 5.46967 3.53033L11.9393 10Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
  <div
    data-testid="second-step"
  >
    second
  </div>
</div>
`;
